rclcpp/test/rclcpp/test_service_introspection.cpp

```diff
@@ -124,10 +124,10 @@ TEST_F(TestServiceIntrospection, service_introspection_nominal)
   ASSERT_THAT(
     client_gid_arr,
     testing::Eq(event_map[ServiceEventInfo::REQUEST_SENT]->info.client_gid));
-  // TODO(@fujitatomoya): Remove this if statement once other rmw implementations support test.
-  // Only rmw_connextdds can pass this test requirement for now.
+  // TODO(@fujitatomoya): Remove this if statement once rmw implementations support test.
+  // rmw_cyclonedds_cpp does not pass this test requirement for now.
   // See more details for https://github.com/ros2/rmw/issues/357
-  if (std::string(rmw_get_implementation_identifier()).find("rmw_connextdds") == 0) {
+  if (std::string(rmw_get_implementation_identifier()).find("rmw_cyclonedds_cpp") != 0) {
```

Comment:
with https://github.com/ros2/rmw_fastrtps/pull/781, `rmw_connextdds` and `rmw_fastrtps` pass this test.

@Yadunund @clalancette @ahcorde  i am not sure about `rmw_zenoh` though, `rmw_zenoh` should skip this test too? if you are not familiar with this, you can let me know. i can check with `rmw_zenoh` too.

Reply:
CC: @Yadunund 

Okay i confirmed with https://github.com/ros2/rmw_zenoh/commit/439d6dc6b0325811793533414dee89196cd72186, `rmw_zenoh` does not have any problem with this test.

<details><summary>colcon test --event-handlers console_direct+ --packages-select rclcpp --ctest-args -R test_service_introspection</summary>

```bash
root@tomoyafujita:~/ros2_ws/colcon_ws# export RMW_IMPLEMENTATION=rmw_zenoh_cpp

root@tomoyafujita:~/ros2_ws/colcon_ws# ros2 run rmw_zenoh_cpp rmw_zenohd &
2024-10-08T19:11:18.926240Z  INFO ThreadId(02) zenoh::net::runtime: Using ZID: d8c755527dbb68b5ad655f6ed52e5a1
2024-10-08T19:11:18.926616Z  INFO ThreadId(02) zenoh::net::runtime::orchestrator: Zenoh can be reached at: tcp/[2607:fd28:130:e15:0:dddd:2:6d]:7447
2024-10-08T19:11:18.926621Z  INFO ThreadId(02) zenoh::net::runtime::orchestrator: Zenoh can be reached at: tcp/43.135.146.89:7447
2024-10-08T19:11:18.926622Z  INFO ThreadId(02) zenoh::net::runtime::orchestrator: Zenoh can be reached at: tcp/[fe80::8d25:6699:7d30:b54c]:7447
Started Zenoh router with id d8c755527dbb68b5ad655f6ed52e5a1.

root@tomoyafujita:~/ros2_ws/colcon_ws# colcon test --event-handlers console_direct+ --packages-select rclcpp --ctest-args -R test_service_introspection
Starting >>> rclcpp
UpdateCTestConfiguration  from :/root/ros2_ws/colcon_ws/build/rclcpp/CTestConfiguration.ini
Parse Config file:/root/ros2_ws/colcon_ws/build/rclcpp/CTestConfiguration.ini
   Site: tomoyafujita
   Build name: (empty)
 Add coverage exclude regular expressions.
Create new tag: 20241008-1912 - Experimental
UpdateCTestConfiguration  from :/root/ros2_ws/colcon_ws/build/rclcpp/CTestConfiguration.ini
Parse Config file:/root/ros2_ws/colcon_ws/build/rclcpp/CTestConfiguration.ini
Test project /root/ros2_ws/colcon_ws/build/rclcpp
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 74
    Start 74: test_service_introspection

74: Test command: /usr/bin/python3 "-u" "/root/ros2_ws/colcon_ws/install/ament_cmake_test/share/ament_cmake_test/cmake/run_test.py" "/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml" "--package-name" "rclcpp" "--output-file" "/root/ros2_ws/colcon_ws/build/rclcpp/ament_cmake_gmock/test_service_introspection.txt" "--command" "/root/ros2_ws/colcon_ws/build/rclcpp/test/rclcpp/test_service_introspection" "--gtest_output=xml:/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml"
74: Working Directory: /root/ros2_ws/colcon_ws/build/rclcpp/test/rclcpp
74: Test timeout computed to be: 60
74: -- run_test.py: invoking following command in '/root/ros2_ws/colcon_ws/build/rclcpp/test/rclcpp':
74:  - /root/ros2_ws/colcon_ws/build/rclcpp/test/rclcpp/test_service_introspection --gtest_output=xml:/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml
74: Running main() from gmock_main.cc
74: [==========] Running 3 tests from 1 test suite.
74: [----------] Global test environment set-up.
74: [----------] 3 tests from TestServiceIntrospection
74: [INFO] [1728414754.103460728] [rmw_zenoh_cpp]: Successfully connected to a Zenoh router with id d8c755527dbb68b5ad655f6ed52e5a1.
74: [ RUN      ] TestServiceIntrospection.service_introspection_nominal
74: [       OK ] TestServiceIntrospection.service_introspection_nominal (24 ms)
74: [ RUN      ] TestServiceIntrospection.service_introspection_enable_disable_events
74: [       OK ] TestServiceIntrospection.service_introspection_enable_disable_events (1005 ms)
74: [ RUN      ] TestServiceIntrospection.service_introspection_enable_disable_event_content
74: [       OK ] TestServiceIntrospection.service_introspection_enable_disable_event_content (2 ms)
74: [----------] 3 tests from TestServiceIntrospection (1032 ms total)
74:
74: [----------] Global test environment tear-down
74: [==========] 3 tests from 1 test suite ran. (1650 ms total)
74: [  PASSED  ] 3 tests.
74: -- run_test.py: return code 0
74: -- run_test.py: inject classname prefix into gtest result file '/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml'
74: -- run_test.py: verify result file '/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml'
1/1 Test #74: test_service_introspection .......   Passed    1.73 sec

The following tests passed:
        test_service_introspection

100% tests passed, 0 tests failed out of 1

Label Time Summary:
gmock     =   1.73 sec*proc (1 test)
mimick    =   1.73 sec*proc (1 test)

Total Test time (real) =   1.73 sec
Finished <<< rclcpp [1.91s]

Summary: 1 package finished [2.94s]
```

</details>